### PR TITLE
[Share] Support v2 android embedder.

### PR DESCRIPTION
**Pull request by cyanglaz**
## Description

Migrate to the new android embedding.

## Related Issues

https://github.com/flutter/flutter/issues/41847

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/share/android/src/main/java/io/flutter/plugins/share/SharePlugin.java [hunk 1]
```diff
@@ -4,59 +4,63 @@
...
   }
 
-  private final Registrar mRegistrar;
+  @Override
+  public void onAttachedToEngine(FlutterPluginBinding binding) {
+    setUpChannel(binding.getFlutterEngine().getDartExecutor());
+  }
+
+  @Override
+  public void onDetachedFromEngine(FlutterPluginBinding binding) {}
```
Comment (matthew-carroll) [inline packages/share/android/src/main/java/io/flutter/plugins/share/SharePlugin.java]: Should we teardown the channel here?

File: packages/share/android/src/main/java/io/flutter/plugins/share/SharePlugin.java [hunk 2]
```diff
@@ -4,59 +4,63 @@
...
-    Intent chooserIntent = Intent.createChooser(shareIntent, null /* dialog title optional */);
-    if (mRegistrar.activity() != null) {
-      mRegistrar.activity().startActivity(chooserIntent);
-    } else {
-      chooserIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-      mRegistrar.context().startActivity(chooserIntent);
-    }
+  private void setUpChannel(BinaryMessenger messenger) {
+    methodChannel = new MethodChannel(messenger, CHANNEL);
+    share = new Share(activity);
```
Comment (matthew-carroll) [inline packages/share/android/src/main/java/io/flutter/plugins/share/SharePlugin.java]: It looks like `activity` is always null at this point. You can probably remove the local `activity` variable and exclusively use the setter method, instead.

File: packages/share/example/android/app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -4,12 +4,19 @@
     <uses-permission android:name="android.permission.INTERNET"/>
 
     <application android:name="io.flutter.app.FlutterApplication" android:label="share_example" android:icon="@mipmap/ic_launcher">
-        <activity android:name=".MainActivity"
+        <activity android:name=".EmbeddingV1Activity"
                   android:launchMode="singleTop"
```
Comment (matthew-carroll) [inline packages/share/example/android/app/src/main/AndroidManifest.xml]: I made a comment about `singleTop` in the sensor PR.

**Comment by cyanglaz** [comment]
@matthew-carroll Updated based on your last review suggestions.

**Comment by matthew-carroll** [comment]
Thanks. Still LGTM!

File: packages/share/pubspec.yaml [hunk 4]
```diff
@@ -25,4 +25,4 @@ dev_dependencies:
 
 environment:
   sdk: ">=2.0.0-dev.28.0 <3.0.0"
-  flutter: ">=1.6.7 <2.0.0"
+  flutter: ">=1.9.1+hotfix.2 <2.0.0"
```
Comment (matthew-carroll) [inline packages/share/pubspec.yaml]: Does the share plugin really require such a recent version?

**Pull request merged**